File: Day-16/Pattern27.py
n = int(input())

def PatternTwentySeven(n):
    i = 1
    while(i <=n):
        space = n-i
        while(space):
            print(' ', end='')
            space-=1
        
        num = i
        printNum = 1
        while(num):
            print(printNum, end='')
            printNum += 1
            num-=1
        
        num2 = i-1
        while(num2):
            print(num2,end='')
            num2-=1
        print(end='\n')
        i +=1

        

        
PatternTwentySeven(n)

File: Day-17/Pattern27.js
function PatternTwentySeven(n) {
  let i = 1;
  while (i <= n) {
    let str = "";
    let spaces = n - i;
    while (spaces) {
      str += " ";
      spaces--;
    }
    let nums = 1;
    let num2 = i - 1;

    while (nums <= i * 2 - 1) {
      if (nums <= i) {
        str += nums;
      } else {
        str += num2;
        num2--;
      }
      nums++;
    }

    console.log(str);
    i++;
  }
}

PatternTwentySeven(4);
